compiler/semstmts.nim

```diff
@@ -446,7 +446,16 @@ proc semVarOrLet(c: PContext, n: PNode, symkind: TSymKind): PNode =
       typ = semTypeNode(c, a.sons[length-2], nil)
 
     var def: PNode = c.graph.emptyNode
+    var compileTimeLet = false
     if a.sons[length-1].kind != nkEmpty:
+      if a.sons[0].kind == nkPragmaExpr:
```

Comment:
Check for ``sfCompileTime in v.flags`` instead?

Reply:
`v` is only analyzed later. Doing a `pragma` scavenger hunt is definitely not the most elegant solution.

Reply:
If it works can it be merged with a TODO saying it needs to be refactored?

Reply:
Doing so is the textbook definition of how to accumulate [technical debt](https://en.wikipedia.org/wiki/Technical_debt), but that's just my opinion.

Reply:
That's a good point. I'll try and come up with a nicer way. I'm also open to suggestions as you know this code better than I do.

Reply:
Does that mean you didn't find a better way? :-/

Reply:
Yes. What I tried didn't work, I'm open to suggestions.

Reply:
evaluate the value of the sym later after we evaluated its pragmas?

Reply:
Ok, I'll try.